modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java

```diff
@@ -130,13 +130,21 @@ public void setBrowserClient(boolean browserClient) {
     }
 
     @Override
-    public String toDefaultValue(Schema p) {
-        if (ModelUtils.isMapSchema(p)) {
+    public String toDefaultValue(Schema schema) {
+        if (ModelUtils.isMapSchema(schema)) {
             return "const {}";
-        } else if (ModelUtils.isArraySchema(p)) {
+        } else if (ModelUtils.isArraySchema(schema)) {
             return "const []";
         }
-        return super.toDefaultValue(p);
+
+        if (schema.getDefault() != null) {
+            if (ModelUtils.isStringSchema(schema)) {
+                return "\"" + schema.getDefault().toString().replaceAll("\"", "\\\"") + "\"";
+            }
+            return schema.getDefault().toString();
+        } else {
+            return "null";
```

Comment:
Quick question: should this be `null` instead of `"null"` similar to https://github.com/OpenAPITools/openapi-generator/pull/6729/files#r446676527 ?

Reply:
I had this discussion on slack. Returning any string causes default value to exist as this string, so returning `"null"` will show in templates that default value is "null", making mustache check `{{{defaultValue}}}` to always be true, thus making all checks for defaultValue pretty much pointless. Returning `null` (not an object) makes `{{{defaultValue}}}` behave as expected in mustache.

Having `toDefaultValue` return `null` allows us to avoid doing `= null` when there is no default value, and doing `bool x = false` when x has default value in .yaml

I tagged you in relevant slack thread.

Reply:
This is in template DartDio - a webframework I don't know, so I made this change backwards compatible for the framework, as they use `int x=  null`. Pure dart2 generator will not have default null.

Reply:
Same thing in DartJaguar, backwards compatible template.